ios/Trifle/Sources/TrifleCertificate.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  KeyHandle.swift
+//  Trifle
+//
+
+import Foundation
+import Wire
+
+public class TrifleCertificate : Codable {
+    private let proto: Certificate
+    
+    public init(data: Data) throws {
+        self.proto = try ProtoDecoder().decode(Certificate.self, from: data)
+    }
+    
+    internal init(from cert: Certificate) throws {
```

Comment:
I believe that the public constructor is the only one we want to expose and implement as part of the library's api. Exposing an internal constructor for the sake of unit tests is a code smell.